routing/http/client/client.go

```diff
@@ -387,24 +389,26 @@ func (c *Client) FindPeers(ctx context.Context, pid peer.ID) (peers iter.ResultI
 		}
 	}()
 
-	var it iter.ResultIter[types.Record]
+	var it iter.ResultIter[types.PeerRecord]
 	switch mediaType {
 	case mediaTypeJSON:
 		parsedResp := &jsontypes.PeersResponse{}
 		err = json.NewDecoder(resp.Body).Decode(parsedResp)
-		var sliceIt iter.Iter[types.Record] = iter.FromSlice(parsedResp.Peers)
+		var sliceIt iter.Iter[types.PeerRecord] = iter.FromSlice(parsedResp.Peers)
 		it = iter.ToResultIter(sliceIt)
 	case mediaTypeNDJSON:
 		skipBodyClose = true
-		it = ndjson.NewRecordsIter(resp.Body)
+		it = ndjson.NewPeerRecordsIter(resp.Body)
 	default:
 		logger.Errorw("unknown media type", "MediaType", mediaType, "ContentType", respContentType)
 		return nil, errors.New("unknown content type")
 	}
 
-	return &measuringIter[iter.Result[types.Record]]{Iter: it, ctx: ctx, m: m}, nil
+	return &measuringIter[iter.Result[types.PeerRecord]]{Iter: it, ctx: ctx, m: m}, nil
 }
 
+// GetIPNS tries to retrieve the [ipns.Record] for the given [ipns.Name]. The record is
+// validated against the given name. If validation fails, an error is returned.
```

Comment:
Is there a specific validation error that's returned if it fails validation that the consumer is supposed to be allowed to check against, or not? If so that error should be called out.

Either way if we're calling out the failed validation -> returned error we may also want to clarify that no record is returned (although that's the standard behavior people implement in Go)

Reply:
> Is there a specific validation error that's returned if it fails validation that the consumer is supposed to be allowed to check against, or not?

No. We could add a generic one in `ipns/` that wraps every validation error.

Reply:
Could add it, but fine as a separate PR